core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/DefaultPackageScanResourceResolver.java

```diff
@@ -97,7 +97,7 @@ protected void findInFileSystem(
             String subPattern)
             throws Exception {
 
-        final ExtendedCamelContext ecc = getCamelContext().adapt(ExtendedCamelContext.class);
+        final ExtendedCamelContext ecc = getCamelContext().getCamelContextExtension();
```

Comment:
Besides the core changes, this is likely what you can see in many of the component changes.